DetectorDescription/Detectors/src/HCalBarrel_geo.cxx

```diff
@@ -110,11 +108,11 @@ static DD4hep::Geometry::Ref_t createHCal (
         dxWedge1, dxWedge2, dzModule, dzModule, drWedge
       ), lcdd.material("Air")
   );
-  for (unsigned int idxLayer = 0; idxLayer < numModulesR; ++idxLayer) {
+  for (unsigned int idxLayer = 1; idxLayer < numModulesR + 1; ++idxLayer) {
     auto layerName = std::string("wedge") + DD4hep::XML::_toString(idxLayer, "layer%d");
     unsigned int sequenceIdx = idxLayer % 2;
     double rminLayer = idxLayer * moduleDimensions.dr();
-    double rmaxLayer = (idxLayer + 1) * moduleDimensions.dr();
+    double rmaxLayer = idxLayer + 1 * moduleDimensions.dr();
```

Comment:
Actually, the +1 is needed here. After having a closer look I realised that when I changed (back in December) the way I construct the components I introduced a bug: I calculate the width of the first layer of modules from the radial position and forgot to add the lower radius of the barrel HCal leading to 0-width modules in the first layer.